packages/presentation/lib/screen/home/home_page.dart

```diff
@@ -0,0 +1,171 @@
+                        : const SizedBox();
+                  },
+                ),
+                const Text(
+                  'Try to predict generated number:',
+                  style: TextStyle(fontSize: 20),
+                ),
+                const SizedBox(height: 15),
+                TextFormField(
+                  maxLength: 1,
+                  controller: _controller,
+                  decoration: const InputDecoration(
+                    border: OutlineInputBorder(),
+                    labelText: 'Number?',
+                  ),
+                  onChanged: (value) {
+                    BlocProvider.of<HomeBloc>(context).add(
+                      ChangeSubmitButtonStateEvent(
+                          isSubmitButtonActive: value.isNotEmpty),
+                    );
+                  },
+                ),
+                const SizedBox(height: 15),
+                Row(
+                  mainAxisAlignment: MainAxisAlignment.center,
+                  children: [
+                    ElevatedButton(
+                        onPressed: () {
+                          BlocProvider.of<HomeBloc>(context)
+                              .add(GenerateNumberEvent());
+                        },
+                        child: const Text('Generate')),
+                    const SizedBox(
+                      width: 15,
+                    ),
+                    BlocSelector<HomeBloc, HomeState, bool>(
+                      selector: (state) => state.isSubmitButtonActive,
+                      builder: (context, isSubmitButtonActive) {
+                        return ElevatedButton(
+                          onPressed: !isSubmitButtonActive
+                              ? null
+                              : () {
+                                  final result = _controller.text;
+                                  final int resultSubmit = int.parse(result);
+                                  BlocProvider.of<HomeBloc>(context).add(
+                                    SubmitNumberEvent(
+                                      predictedNumber: resultSubmit,
+                                    ),
+                                  );
+                                },
+                          child: const Text('Submit'),
+                        );
+                      },
+                    ),
+                  ],
+                ),
+                BlocSelector<HomeBloc, HomeState, int>(
+                  selector: (state) => state.attempts,
+                  builder: (context, attempts) {
+                    return attempts >= 1
+                        ? Text(
+                            'Try again. You have ${3 - attempts} more attempts',
+                            style: const TextStyle(fontSize: 15),
+                          )
+                        : const SizedBox();
+                  },
+                ),
+              ],
+            ),
+          ),
+        ),
+      ),
+    );
```

Comment:
to increase readability it is better to use 'extract flutter widget'

Reply:
what exactly

Reply:
what exactly?

Reply:
up to u